x/oracle/types/genesis.go

```diff
@@ -10,6 +14,10 @@ func DefaultGenesis() *GenesisState {
 // Validate performs basic genesis state validation returning an error upon any
 // failure.
 func (gs GenesisState) Validate() error {
-
+	for i, oracle := range gs.Oracles {
+		if gs.Oracles[i].Address != oracle.Address {
+			return sdkerrors.Wrapf(ErrInvalidGenesisOracle, "oracle address: %s", oracle.GetAddress())
+		}
+	}
 	return nil
```

Comment:
Hm. I think this logic always returns true.

Reply:
Umm. What part of genesis are you trying to validate?
Is the address valid?
Or is it a duplicate stored oracle address?

Reply:
The address. So I changed to map for genesis state. But the this logic always returns true. 😂 
So I can change like this.
```go
func (gs GenesisState) Validate() error {
	for _, oracle := range gs.Oracles {
		oracleAddr, err := sdk.AccAddressFromBech32(oracle.Address)
		
		if err != nil {
			return sdkerrors.Wrapf(ErrInvalidGenesisOracleAddr, "oracle address: %s", oracleAddr.String())
		}
	}
	return nil
}
```
How about this?

Reply:
cd3b9e1